main/lsp/LSPTypechecker.cc

```diff
@@ -373,7 +387,7 @@ LSPTypechecker::SlowPathResult LSPTypechecker::runSlowPath(LSPFileUpdates &updat
                     if (!updates.updatedFiles.empty()) {
                         for (auto &ast : updates.updatedFileIndexes) {
                             if (ast.tree) {
-                                updates.updatedFinalGSFileIndexes.emplace_back(std::move(ast));
+                                this->indexedFinalGS[ast.file.id()] = std::move(ast);
```

Comment:
Is it worth copying the spirit of the comment we deleted, "Before making preemption or cancelation possible"

something about how in this spot we're "committing" the tree by writing into the map? Is that an accurate summary of what this line is doing?

Reply:
I think we could actually skip this step if we wanted to, and instead rely on the fast path to repopulate the `indexedFinalGS` cache. (I called this out as an option on #8738 as an alternative to looking through the indexed trees for the ones we know are open in the editor.)

We could put a comment in, but I think it's less important than it was before: we would previously be modifying the in-memory cache of indexed trees and those changes would be recorded in the `UndoState` so that they could be rolled back. When we had to maintain the connection between the indexer's `GlobalState` and those indexed trees, it was pretty critical that we had updated those trees before starting the slow path, as we needed to ensure that our name table was sane.